lib/DistributionProvider.pm

```diff
@@ -35,13 +35,13 @@ If there is no matched version, then returns Tumbleweed as the default one.
 =cut
 
 sub provide {
+    return Distribution::AgamaDevel->new() if get_var('VERSION', '') =~ /agama/;
```

Comment:
most likely we will need here `is_opensuse` and another line `is_sle` or similar.

Reply:
Could we check 'FLAVOR' instead of 'VERSION'? Because for Leap it is hard to distinguish agama test since version = '16.0'.  If flavors with 'agama' in name are only used for agama test purpose, then this can be used for both osd and o3. WDYT?   
Test with this condition for leap test: https://openqa.opensuse.org/tests/4511165#details

Reply:
Lemon can check  by `is leap 16` after line 42, and you can leave this condition where it was, so we sort of order the matching, first sle products, later opensuse ones then those fictition ones we have for devel (that is why that ugly regex for now) and finally tw.

Reply:
Ok, it works. https://openqa.opensuse.org/tests/4511253#details  So we make changes in the same PR, or I submit another PR for Leap?

Reply:
5b383abf040e62338fbdcf8380f91a2efe572ec5